### PR TITLE
refactor: refactor and optimize extra properties serializer

### PR DESCRIPTION
**Pull request by ivan-tymoshenko**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: index.js [hunk 1]
```diff
@@ -228,92 +228,78 @@ function inferTypeByKeyword (schema) {
   return schema.type
 }
 
-function addPatternProperties (location) {
+function buildExtraObjectPropertiesSerializer (location) {
   const schema = location.schema
-  const pp = schema.patternProperties
+  const propertiesKeys = Object.keys(schema.properties || {})
```
Comment (Uzlopak) [inline index.js:233]: at this point we have the length of properties. What if it is empty? So could we skip stuff?

Comment (ivan-tymoshenko) [reply index.js:233]: No, the Keys properties are needed here in order not to add the same property twice. If the property was already serialized as a regular property, we should not serialize it as a pattern or additional property. So if the user passes an empty `property` value, the propertiesKeys would be an empty array, and all other values would be added as pattern or additional values.

File: index.js [hunk 2]
```diff
@@ -228,92 +228,72 @@ function inferTypeByKeyword (schema) {
...
-      pp[regex] = ppLocation.schema
-    }
+  const patternPropertiesSchema = patternPropertiesLocation.schema
 
-    try {
-      RegExp(regex)
-    } catch (err) {
-      throw new Error(`${err.message}. Found at ${regex} matching ${JSON.stringify(pp[regex])}`)
-    }
+  if (patternPropertiesSchema !== undefined) {
```
[the remark this answers is not in the document]
Comment (ivan-tymoshenko) [reply index.js:249]: I think it should be done in schema validator.
https://github.com/fastify/fast-json-stringify/blob/master/lib/schema-validator.js

[the remark this answers is not in the document]
Comment (Uzlopak) [reply index.js:249]: Ah i understand.

**Comment by ivan-tymoshenko** [comment]
@mcollina Can we merge this?

**Review by Eomm — APPROVED** [review]

**Pull request merged**